File: backend/agentic_crm_sdk/deals.py
"""
Deals API client module.
"""
from agentic_crm_sdk.client import AgenticCRM


class DealsClient:
    def __init__(self, client: AgenticCRM):
        self._client = client

    def list(self, **params):
        return self._client.get("/api/deals", params=params)

    def get(self, deal_id: int):
        return self._client.get(f"/api/deals/{deal_id}")

    def create(self, data: dict):
        return self._client.post("/api/deals", json=data)

    def update(self, deal_id: int, data: dict):
        return self._client.patch(f"/api/deals/{deal_id}", json=data)

    def delete(self, deal_id: int):
        return self._client.delete(f"/api/deals/{deal_id}")

    def activities(self, deal_id: int):
        return self._client.get(f"/api/deals/{deal_id}/activities")

File: frontend/lib/sdk/deals.ts
import type { Deal, DealStage } from '../api'
import { AgenticCRM } from './client'

export interface CreateDealInput {
  contact_id: number
  company_id?: number
  name: string
  value: number
  stage?: DealStage
  expected_close_date?: string
  notes?: string
}

export interface UpdateDealInput {
  name?: string
  value?: number
  stage?: DealStage
  expected_close_date?: string
  actual_close_date?: string
  notes?: string
}

export class DealsClient {
  constructor(private client: AgenticCRM) {}

  async list(params?: { stage?: DealStage }): Promise<Deal[]> {
    return this.client.get<Deal[]>('/api/deals', params as Record<string, string>)
  }

  async get(dealId: number): Promise<Deal> {
    return this.client.get<Deal>(`/api/deals/${dealId}`)
  }

  async create(data: CreateDealInput): Promise<Deal> {
    return this.client.post<Deal>('/api/deals', data)
  }

  async update(dealId: number, data: UpdateDealInput): Promise<Deal> {
    return this.client.patch<Deal>(`/api/deals/${dealId}`, data)
  }

  async delete(dealId: number): Promise<void> {
    return this.client.delete<void>(`/api/deals/${dealId}`)
  }

  async activities(dealId: number): Promise<unknown[]> {
    return this.client.get<unknown[]>(`/api/deals/${dealId}/activities`)
  }
}